apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ApolloCompiler.kt

```diff
@@ -170,6 +170,11 @@ object ApolloCompiler {
       options.operationOutputFile.writeText(operationOutput.toJson("  "))
     }
 
+    // Merge scalar mappings from upstream modules and this module's options
+    val allScalarMapping = options.incomingCompilerMetadata.fold(options.scalarMapping.toMutableMap()) { acc, it ->
+      acc.apply { putAll(it.scalarMapping) }
```

Comment:
This feels dangerous. What if there's a conflicting mapping in 2 modules? Should we mandate that all scalar mappings are defined in the schema module? 

I can't think of any use case where different modules could use different mappings but if something comes up we can always relax the restriction.

Reply:
I thought it was already the case actually. The multi module [doc](https://www.apollographql.com/docs/kotlin/advanced/multi-modules/) says "The schema module and only the schema module must define customScalarsMapping".

Should I ensure there are no duplicates and throw if there are any? Or should we not merge them at all (not sure how that would work)?

Reply:
Better throw fast if we see something like this.